contracts/events/TopicEvent.sol

```diff
@@ -299,54 +292,52 @@ contract TopicEvent is ITopicEvent, BaseContract, Ownable {
     }
 
     /* 
-    * @notice Calculates the QTUM tokens won based on the sender's QTUM token contributions.
-    * @return The amount of QTUM tokens won.
+    * @notice Calculates the BOT and QTUM tokens won based on the sender's contributions.
+    * @return The amount of BOT and QTUM tokens won.
     */
-    function calculateQtumContributorWinnings() 
+    function calculateWinnings()
         public 
         view
         inCollectionStatus()
-        returns (uint256)  
+        returns (uint256, uint256)  
     {
-        uint256 senderContribution = balances[resultIndex].bets[msg.sender];
-        uint256 winnersTotal = balances[resultIndex].totalBets;
-        uint256 losersTotalMinusCut = 0;
-        for (uint8 i = 0; i < numOfResults; i++) {
-            if (i != resultIndex) {
-                losersTotalMinusCut = losersTotalMinusCut.add(balances[i].totalBets);
-            }
-        }
-        losersTotalMinusCut = losersTotalMinusCut.mul(90).div(100);
-        uint256 senderContributionMinusCut = senderContribution.mul(90).div(100);
-        return senderContribution.mul(losersTotalMinusCut).div(winnersTotal).add(senderContributionMinusCut);
-    }
+        uint256 votes = balances[resultIndex].votes[msg.sender];
+        uint256 bets = balances[resultIndex].bets[msg.sender];
 
-    /*
-    * @notice Calculates the QTUM and BOT tokens won based on the sender's BOT contributions.
-    * @return The amount of QTUM and BOT tokens won.
-    */
-    function calculateBotContributorWinnings()
-        public
-        view
-        inCollectionStatus()
-        returns (uint256, uint256) 
-    {
-        // Calculate BOT won
-        uint256 senderContribution = balances[resultIndex].votes[msg.sender];
-        uint256 winnersTotal = balances[resultIndex].totalVotes;
+        // Calculate Qtum reward total
         uint256 losersTotal = 0;
         for (uint8 i = 0; i < numOfResults; i++) {
```

Comment:
Why this for loop is outside of if (bets>0) while votes one is inside? losersTotal is used by rewardQtum which is needed for qtumWon calculation in votes, so qtumWon and rewardQtum should be initialized outside of if(), but calculation should be inside.

Reply:
For example, for all losers they don't need this for loop but will go through it anyway.

Reply:
Community asked a very interesting question and Sam raised it. I post it here. What would happen if there's zero winning bet but a lot of losing bet. Losers will get zero from line 1 so they can't withdraw anything, and there's no winner to withdraw losers money.

Reply:
1&2. If you look at the logic, calculating `botWon` needs `rewardQtum`. And `losersTotal` is needed to calculate `rewardQtum` so that is why they are outside the `if` block.

If I kept this logic inside the `if` block, if a user only had `votes > 0`, `rewardQtum` would be 0 and `qtumWon` would be wrong.

3. That is a good point. Let's bring this up at tomorrows meeting. Losers still should not be able to get their coins back since they lost fair and square.